include/kernel.h

```diff
@@ -1445,6 +1445,24 @@ extern void k_queue_merge_slist(struct k_queue *queue, sys_slist_t *list);
 extern void *k_queue_get(struct k_queue *queue, s32_t timeout);
 
 /**
+ * @brief Remove an element from a queue.
+ *
+ * This routine removes data item from @a queue. The first 32 bits of the
+ * data item are reserved for the kernel's use.
```

Comment:
Or we changed it to use sys_dlist_t as the wait_q, though this can be done separately if we find that would be a problem.